include/arch/arm/arch/32/mode/hardware.h

```diff
@@ -85,6 +85,10 @@
 #ifndef __ASSEMBLER__
 /* It is required that USER_TOP must be aligned to at least 20 bits */
 compile_assert(USER_TOP_correctly_aligned, IS_ALIGNED(USER_TOP, 20));
+/* It is required on arm_hyp that USER_TOP isn't lower than the top GiB */
```

Comment:
Where does this requirement come from?

Reply:
https://github.com/seL4/seL4/blob/master/src/arch/arm/32/kernel/vspace.c#L316.  The kernel has it's own PL2 address space, but only uses the top 1 GiB.